python/paddle/fluid/tests/unittests/test_while_loop_op.py

```diff
@@ -110,24 +113,146 @@ def internal_body(j, init, sums):
+            ten = layers.fill_constant(shape=[1], dtype='int64', value=10)
+            one = layers.fill_constant(shape=[1], dtype='float32', value=1)
+            x = layers.fill_constant(shape=[1], dtype='float32', value=0)
+            x.stop_gradient = False
+
+            out = layers.while_loop(cond, body, [i, x])
+            mean = layers.mean(out[1])
+            append_backward(mean)
+
+        place = fluid.CUDAPlace(0) if core.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
+        exe = fluid.Executor(place)
+
+        feed_x = np.asarray([10]).astype('float32')
+        res = exe.run(program, feed={}, fetch_list=[mean.name, mean.grad_name])
+        x_grad = np.ones(1).astype('float32')
+        self.assertTrue(np.allclose(np.asarray(res[0]), feed_x))
+        self.assertTrue(np.allclose(np.asarray(res[1]), x_grad))
+
+
+class TestApiWhileLoop_NestedWithBackward(unittest.TestCase):
+    def test_nested_net_with_backward(self):
+        def cond_outside(i, x):
+            return layers.less_than(i, ten)
+
+        def body_outside(i, x):
+            def cond_inside(i, x):
+                return layers.less_than(i, five)
+
+            def body_inside(i, x):
+                x = layers.elementwise_add(x=x, y=one)
+                i = layers.increment(i)
+                return [i, x]
+
+            temp = layers.while_loop(cond_inside, body_inside, [i, x])
+            x = layers.elementwise_add(x=temp[1], y=two)
+            i = layers.increment(i)
+            return [i, x]
+
+        main_program = fluid.default_main_program()
+        startup_program = fluid.default_startup_program()
+
+        with fluid.program_guard(main_program, startup_program):
+            i = layers.fill_constant(shape=[1], dtype='int64', value=0)
+            ten = layers.fill_constant(shape=[1], dtype="int64", value=10)
+            five = layers.fill_constant(shape=[1], dtype="int64", value=5)
+            two = layers.fill_constant(shape=[1], dtype="float32", value=2)
+            one = layers.fill_constant(shape=[1], dtype="float32", value=1)
+            x = layers.fill_constant(shape=[1], dtype="float32", value=1)
+            x.stop_gradient = False
+            out = layers.while_loop(cond_outside, body_outside, [i, x])
+
+            mean = layers.mean(out[1])
+            append_backward(mean)
+
+        place = fluid.CUDAPlace(0) if core.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
+        exe = fluid.Executor(place)
+        res = exe.run(main_program,
+                      feed={},
+                      fetch_list=[mean.name, mean.grad_name])
+
+        feed_x = np.asarray([16]).astype("float32")
+        x_grad = np.ones(1).astype("int32")
+        self.assertTrue(np.allclose(np.asarray(res[0]), feed_x))
+        self.assertTrue(np.allclose(np.asarray(res[1]), x_grad))
+
+
+class TestApiWhileLoopWithSwitchCase(unittest.TestCase):
+    def test_with_(self):
```

Comment:
This name "test_with_" confused me. You may change it into "test_with_switch_case"?

Reply:
Done.